particles/Chooser/Chooser.js

```diff
@@ -26,32 +26,52 @@
         this.emit('values', result);
       });
       this.when([new StateChanges('values'), new SlotChanges()], async e => {
-        let values = this.states.get('values');
-        if (values.length <= 0) {
-          this.releaseSlot('action');
+        let slotName = 'action';
```

Comment:
this.spec.renders is the Map of slots that the particle renders. maybe it should be used instead of hardcoded slot name?
not necessarily in this change, but just a thought.